admin/c2cgeoportal_admin/routes.py

```diff
@@ -39,18 +39,33 @@ def includeme(config):
     )
     from c2cgeoportal_commons.models.static import User
 
+    visible_routes = [
+        ("themes", Theme),
+        ("layer_groups", LayerGroup),
+        ("layers_wms", LayerWMS),
+        ("layers_wmts", LayerWMTS),
+        ("ogc_servers", OGCServer),
+        ("restriction_areas", RestrictionArea),
+        ("users", User),
+        ("roles", Role),
+        ("functionalities", Functionality),
+        ("interfaces", Interface),
+    ]
+
+    admin_interface_config = config.registry.settings["admin_interface"]
+
+    # Include pages
+    for url_path, model in admin_interface_config.get("include_pages", []):
+        mytuple = (url_path, config.maybe_dotted(model))
+        visible_routes.append(mytuple)
+
+    # Exclude pages
+    for url_path in admin_interface_config.get("exclude_pages", []):
+        for i, route in enumerate(visible_routes):
```

Comment:
you delete lines of visible_routes inside the enumeration
pretty sure this will remove bad route if you remove more that one, as indexes will change inside the iteration.

You're old way to do it (by constructing a new list)  was good: [(path, model) for (path, model) in visible_routes if path not in exclude_pages]

Or you can use `filter`